File: cpp/16.06.19/ifelse.cpp
#include <iostream>
using namespace std;

int main(int argc, char** argv)
{
    int e1s = 0;
    int e1e = 100; 
    int e2s = 14;
    int e2e = 18;
    if (e1e < e2s || e2e < e1s)
    {
        cout << "No Overlap" << endl;
    }
    else
    {
        if ((e2s > e1s && e2e < e1e) || (e1s > e2s && e1e < e2e))
        {
            cout << "Full Overlap"<< endl;
        }
        else
        {
            cout << "Partial Overlap" << endl;
        }
        
    }
}
// Reviewed by Amrit

File: java/16.06.19/ifelse.java
import java.util.*;

public class ifelse
{
    public static void main(String[] args)
    {
        int e1s = 0;
        int e1e = 10;
        int e2s = 3;
        int e2e = 13;

        if (e1e < e2s || e2e < e1s)
        {
            System.out.println("No Overlap");
        }
        else
        {
            if((e2s > e1s && e2e < e1e) || (e1s > e2s && e1e < e2e))
            {
                System.out.println("Full Overlap");
            }
            else
            {
                System.out.println("Patial Overlap");
            }
        }
    }
}
// Reviewed by Amrit


